examples/Protobuf/messages/CMakeLists.txt

```diff
@@ -1,7 +1,25 @@
+set(MESSAGES_PROTOS messages.proto)
+set(MESSAGES_SRCS)
+set(MESSAGES_HDRS)
 
-# required since no linking
-include_directories(${PROTOBUF_INCLUDE_DIR})
-PROTOBUF_GENERATE_CPP(MESSAGES_SRC MESSAGES_HEADER messages.proto)
-add_library(messages ${MESSAGES_HEADER} ${MESSAGES_SRC})
+#Code Generation
+foreach(proto_file ${MESSAGES_PROTOS})
+  get_filename_component(proto_file_abs ${proto_file} ABSOLUTE)
+  get_filename_component(basename ${proto_file} NAME_WE)
+  set(generated_files ${basename}.pb.cc ${basename}.pb.h)
 
+  list(APPEND MESSAGES_SRCS ${basename}.pb.cc)
+  list(APPEND MESSAGES_HDRS ${CMAKE_CURRENT_BINARY_DIR}/${basename}.pb.h)
 
+  add_custom_command(
+      OUTPUT ${generated_files}
+      COMMAND protobuf::protoc
+      ARGS --cpp_out ${CMAKE_CURRENT_BINARY_DIR} -I ${CMAKE_CURRENT_SOURCE_DIR} ${proto_file_abs}
```

Comment:
We don't need `ARGS`. From [documentation](https://cmake.org/cmake/help/latest/command/add_custom_command.html):

```
The optional ARGS argument is for backward compatibility and will be ignored
```